en/rest/hooks.mdown

```diff
@@ -0,0 +1,670 @@
+
+```bash
+curl -X GET \
+  -H "X-Parse-Application-Id: ${APPLICATION_ID}" \
+  -H "X-Parse-Master-Key: ${MASTER_KEY}" \
+  -H "Content-Type: application/json" \
+  https://api.parse.com/1/hooks/functions
+```
+```python
+import json,httplib
+connection = httplib.HTTPSConnection('api.parse.com', 443)
+connection.connect()
+connection.request('GET', '/1/hooks/functions', '', {
+       "X-Parse-Application-Id": "${APPLICATION_ID}",
+       "X-Parse-Master-Key": "${MASTER_KEY}",
+       "Content-Type": "application/json"
+     })
+result = json.loads(connection.getresponse().read())
+print result
+```
+
+The output may look like this:
+```json
+{
+  "results": [
+    { "functionName": "sendMessage", "url": "https://api.example.com/sendMessage" },
+    { "functionName": "sendMessage" },
+    { "functionName": "foo", "url": "https://api.example.com/foo" },
+    { "functionName": "bar" }
+  ]
+}
+```
+The output is a json object with one key: "results" whose value is a list of cloud code functions
+or cloud code function webhooks.
+
+To fetch a single cloud function with a given name, use:
+
+```bash
+curl -X GET \
+  -H "X-Parse-Application-Id: ${APPLICATION_ID}" \
+  -H "X-Parse-Master-Key: ${MASTER_KEY}" \
+  -H "Content-Type: application/json" \
+  https://api.parse.com/1/hooks/functions/sendMessage
+```
+```python
+import json,httplib
+connection = httplib.HTTPSConnection('api.parse.com', 443)
+connection.connect()
+connection.request('GET', '/1/hooks/functions/sendMessage', '', {
+       "X-Parse-Application-Id": "${APPLICATION_ID}",
+       "X-Parse-Master-Key": "${MASTER_KEY}",
+       "Content-Type": "application/json"
+     })
+result = json.loads(connection.getresponse().read())
+print result
+```
+
+The output is a json object with one key: "results" whose value is a list of cloud code function
+or cloud code function webhooks with the given name.
+
+```json
+{
+  "results": [
+    { "functionName": "sendMessage", "url": "https://api.example.com/sendMessage" },
+    { "functionName": "sendMessage" },
+  ]
+}
+```
+
+If only one of cloud code function or cloud code trigger function with the given name exists,
+then only that one value will be present in the list.
+
+```bash
+curl -X GET \
+  -H "X-Parse-Application-Id: ${APPLICATION_ID}" \
+  -H "X-Parse-Master-Key: ${MASTER_KEY}" \
+  -H "Content-Type: application/json" \
+  https://api.parse.com/1/hooks/functions/foo
+```
+```python
+import json,httplib
+connection = httplib.HTTPSConnection('api.parse.com', 443)
+connection.connect()
+connection.request('GET', '/1/hooks/functions/foo', '', {
+       "X-Parse-Application-Id": "${APPLICATION_ID}",
+       "X-Parse-Master-Key": "${MASTER_KEY}",
+       "Content-Type": "application/json"
+     })
+result = json.loads(connection.getresponse().read())
+print result
+```
+
+The output may look like this:
+
+```json
+{
+  "results": [
+    { "functionName": "foo", "url": "https://api.example.com/foo" }
```

Comment:
If in the previous section we explain that the result item with the url is the webhook, the other is the cloud function, and either could be missing, I don't think we actually need this example.